### PR TITLE
Autoupgrade

### PR DESCRIPTION
**Pull request by gysmo38**
I update autoupdate script to create a version file with date, branch and commit.
Before upgrading it check if the VERSION file exist if commit id is equal to remote commit id, it not try to upgrade.
After I will try to use commit id to improve upgrade process. Instead of check all files, just check updated files.

### CODE REVIEW

**Comment by gysmo38** [comment]
Please not accept for the moment. I will made a change on autoupdate script. It will increase the speed of upgrade :)

**Comment by gysmo38** [comment]
My PR is finish. You can review it for commit. 
First update it create a VERSION file with, date, branch and commit version. Next update it check VERSION file and compare commit with github last commit. If this is equal it do nothing if not, it upgrade only changed files. The upgrade process is faster :)

File: firmware_mod/autoupdate.sh [hunk 1]
```diff
@@ -235,11 +244,22 @@ fi
...
 
-
-
-log "Getting list of remote files."
-FIRST=$(${CURL} -s ${GITHUBURL}/${REPO}/contents/${REMOTEFOLDER}?ref=${BRANCH})
-FILES=$(getfiles "${FIRST}")
+if [ -f "$VERSION_FILE" ]; then
+    LOCALCOMMITID=$(${JQ} -r .commit ${VERSION_FILE})  
+    if [ ${LOCALCOMMITID} = ${REMOTECOMMITID} ]; then
+        logerror "You have already last version"
```
Comment (jmtatsch) [inline firmware_mod/autoupdate.sh:250]: latest version

File: firmware_mod/autoupdate.sh [hunk 2]
```diff
@@ -221,6 +224,12 @@ done
 
 log "Starting AutoUpdate on branch ${BRANCH}"
 
+######################################################""
+# Get date and las commit ID from Github
```
Comment (jmtatsch) [inline firmware_mod/autoupdate.sh:228]: Last commit ID

File: firmware_mod/www/update.html [hunk 3]
```diff
@@ -94,11 +109,16 @@
     <header class='card-header'>
         <p class='card-header-title'>Update</p>
     </header>
-    <div class='card-content'>
+    <div class='card-content' style="display: none; visibility: hidden;" id='no_updatebox'>
+        <div class="field is-horizontal">
+            <p>You have already the last version.</p>
```
Comment (jmtatsch) [inline firmware_mod/www/update.html:114]: Latest

File: firmware_mod/www/cgi-bin/state.cgi [hunk 4]
```diff
@@ -150,9 +150,15 @@ if [ -n "$F_cmd" ]; then
...
-  version)
-    echo $(cat /system/sdcard/.lastCommitDate);
-    ;;
+  version)   
+    if [ -s "/system/sdcard/VERSION" ]; then   
+    V_BRANCH=$(/system/sdcard/bin/jq -r .branch /system/sdcard/VERSION)
+    V_COMMIT=$(/system/sdcard/bin/jq -r .commit /system/sdcard/VERSION)
+    echo "commit <b>${V_COMMIT}</b> from branch <b>${V_BRANCH}</b>"
+    else                                    
+       echo "No version file. Need upgrade"
```
Comment (jmtatsch) [inline firmware_mod/www/cgi-bin/state.cgi:159]: Need to update to create a version file.

**Comment by nik0** [comment]
> Except for some typos seems good to me. @nik0 you originally contributed the updating function, what do you think?

I didn't test it but it seems ok for me

**Comment by gysmo38** [comment]
I update the PR with typo fixies.
@nik0 You will see, the upgrade with version file go really faster and avoid to reach limit with anonymous github api.

**Pull request merged**